src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -2194,13 +2195,20 @@ public override bool HandleCall(MethodIL callingMethodBody, MethodDesc calledMet
                                 }
 
                                 comDangerousMethod |= IsComInterop(marshalAsDescriptor, calledMethod.Signature[paramIndex]);
+                                abstractDelegatesMethod |= IsAbstractDelegatesInterop(marshalAsDescriptor, calledMethod.Signature[paramIndex]);
                             }
 
                             if (comDangerousMethod)
                             {
                                 reflectionContext.AnalyzingPattern();
                                 reflectionContext.RecordUnrecognizedPattern(2050, $"P/invoke method '{calledMethod.GetDisplayName()}' declares a parameter with COM marshalling. Correctness of COM interop cannot be guaranteed after trimming. Interfaces and interface members might be removed.");
                             }
+
+                            if (abstractDelegatesMethod)
+                            {
+                                reflectionContext.AnalyzingPattern();
+                                reflectionContext.RecordUnrecognizedPattern(3055, $"P/invoke method '{calledMethod.GetDisplayName()}' declares a parameter with abstract delegate. Correctness of interop for abstract delegates cannot be guaranteed after trimming. Delegates might be removed.");
```

Comment:
> Delegates might be removed.

I do not think that's a correct statement. The actual delegate type won't be removed by trimming. 

The problem is that AOT compiler may not generate the interop marshaling code for delegate.

Reply:
MIchal fix that messaging in resources files. I will use resources and as such this text will go away.